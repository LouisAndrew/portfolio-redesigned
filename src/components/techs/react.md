---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
