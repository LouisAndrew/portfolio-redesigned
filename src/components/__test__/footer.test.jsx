import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Footer from '../footer'

describe('Footer element on every layout renders correctly', () => {

        afterEach( cleanup )

        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render(<Footer />, div)
        })

        it('renders correctly', () => {

                const { getByTestId } = render(<Footer />)
        })

        it('matches snapshot', () => {

                const tree = renderer.create(<Footer />).toJSON()
                expect(tree).toMatchSnapshot()
        })

})