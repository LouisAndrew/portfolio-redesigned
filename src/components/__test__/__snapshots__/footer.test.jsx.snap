// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer element on every layout renders correctly matches snapshot 1`] = `
<footer
  className="sc-AxjAm kAilxx wrap"
>
  <div
    className="wrapper"
  >
    <ul
      className="sc-AxirZ ejkNpX"
      data-testid="links-wrapper"
    >
      <li>
        <a
          href="/#hero"
        >
          HOME
        </a>
      </li>
      <li>
        <a
          href="/#about"
        >
          ABOUT
        </a>
      </li>
      <li>
        <a
          href="/#contact"
        >
          CONTACT
        </a>
      </li>
      <li>
        <a
          href="/projects"
        >
          PROJECTS
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
